src/lib.rs

```diff
@@ -50,6 +56,19 @@ extern crate failure;
 extern crate serde_derive;
 extern crate termcolor;
 
+#[cfg(target_os = "windows")]
+#[path = "windows_window.rs"]
+pub mod window;
```

Comment:
Please refactor these modules. There should be a `gui` or `window` module which contains `win.rs` and `x11.rs`. Gate the loading of `window`(or `gui`) via the feature mentioned above, then select `win` or `x11` based on platform